lib/FrontendTool/ReferenceDependencies.cpp

```diff
@@ -278,98 +294,73 @@ bool swift::emitReferenceDependencies(DiagnosticEngine &diags, SourceFile *SF,
+  out << dynamicLookupProviders << ":\n";
+  class NameCollector : public VisibleDeclConsumer {
+  private:
+    SmallVector<DeclBaseName, 16> names;
+
+  public:
+    void foundDecl(ValueDecl *VD, DeclVisibilityKind Reason) override {
+      names.push_back(VD->getBaseName());
+    }
+    ArrayRef<DeclBaseName> getNames() {
+      llvm::array_pod_sort(
+          names.begin(), names.end(),
+          [](const DeclBaseName *lhs, const DeclBaseName *rhs) {
+            return lhs->compare(*rhs);
+          });
+      names.erase(std::unique(names.begin(), names.end()), names.end());
+      return names;
     }
-  }
-
-  auto sortedByName =
-      [](const llvm::DenseMap<DeclBaseName, bool> map) ->
-        SmallVector<std::pair<DeclBaseName, bool>, 16> {
-    SmallVector<std::pair<DeclBaseName,bool>, 16> pairs{map.begin(), map.end()};
-    llvm::array_pod_sort(pairs.begin(), pairs.end(),
-                         [](const std::pair<DeclBaseName, bool> *first,
-                            const std::pair<DeclBaseName, bool> *second) -> int{
-      return first->first.compare(second->first);
-    });
-    return pairs;
   };
+  NameCollector collector;
+  SF->lookupClassMembers({}, collector);
+  for (DeclBaseName name : collector.getNames()) {
+    out << "- \"" << escape(name) << "\"\n";
+  }
+}
 
-  const ReferencedNameTracker *const tracker = SF->getReferencedNameTracker();
-  assert(tracker && "Cannot emit reference dependencies without a tracker");
-
-  out << "depends-top-level:\n";
-  for (auto &entry : sortedByName(tracker->getTopLevelNames())) {
+static void writeTopLevelDependencies(
+    const ArrayRef<std::pair<DeclBaseName, bool>> sortedTopLevelNames,
+    llvm::raw_fd_ostream &out) {
+  out << topLevelDependencies << ":\n";
+  for (auto &entry : sortedTopLevelNames) {
     assert(!entry.first.empty());
     out << "- ";
     if (!entry.second)
       out << "!private ";
     out << "\"" << escape(entry.first) << "\"\n";
   }
+}
 
-  out << "depends-member:\n";
-  auto &memberLookupTable = tracker->getUsedMembers();
-  using TableEntryTy = std::pair<ReferencedNameTracker::MemberPair, bool>;
-  std::vector<TableEntryTy> sortedMembers{
-    memberLookupTable.begin(), memberLookupTable.end()
-  };
-  llvm::array_pod_sort(sortedMembers.begin(), sortedMembers.end(),
-                       [](const TableEntryTy *lhs,
-                          const TableEntryTy *rhs) -> int {
-    if (lhs->first.first == rhs->first.first)
-      return lhs->first.second.compare(rhs->first.second);
-
-    if (lhs->first.first->getName() != rhs->first.first->getName())
-      return lhs->first.first->getName().compare(rhs->first.first->getName());
-
-    // Break type name ties by mangled name.
-    auto lhsMangledName = mangleTypeAsContext(lhs->first.first);
-    auto rhsMangledName = mangleTypeAsContext(rhs->first.first);
-    return lhsMangledName.compare(rhsMangledName);
-  });
+using TableEntryTy = std::pair<ReferencedNameTracker::MemberPair, bool>;
```

Comment:
I suppose I'm guilty of this in the original function too, but this isn't the entry type for *every* table, only the "members" table. "MemberTableEntryTy"?

Reply:
Thanks for the concrete suggestion. Fixed.